src/util/check-ref-resource.ts

```diff
@@ -0,0 +1,43 @@
+/*-----------------------------------------------------------------------------------------------
+ *  Copyright (c) Red Hat, Inc. All rights reserved.
+ *  Licensed under the MIT License. See LICENSE file in the project root for license information.
+ *-----------------------------------------------------------------------------------------------*/
+
+import { cli } from '../cli';
+import { Command } from '../cli-command';
+import { ContextType } from '../context-type';
+import { PipelineTaskRunData } from '../tekton';
+import { TektonNode } from '../tree-view/tekton-node';
+import * as vscode from 'vscode';
+
+export async function getTaskRunResourceList(): Promise<PipelineTaskRunData[]> {
+  const result = await cli.execute(Command.listTaskRun());
+  if (result.error) {
+    // ignore
+  }
+  let data: PipelineTaskRunData[] = [];
+  try {
+    const r = JSON.parse(result.stdout);
+    data = r.items ? r.items : data;
+  } catch (ignore) {
+    // ignore
+  }
+  return data;
+}
+  
+export function referenceOfTaskAndClusterTaskInCluster(item: TektonNode, taskRunList: PipelineTaskRunData[]): boolean{
+  if (taskRunList.length !== 0 && (item.contextValue === ContextType.TASK || item.contextValue === ContextType.CLUSTERTASK)) {
+    const found = taskRunList.some((value) => {
+      if (value.spec.taskRef.name === item.getName()) {
```

Comment:
By using `(item.contextValue === ContextType.TASK || item.contextValue === ContextType.CLUSTERTASK)` above you are not really checking the type of the resource. If i have a task and a clustertask with the same name this will return always true even if one of the two is not used. I think there should be a check here as well.